lib/solidus_kustomer/client.rb

```diff
@@ -34,14 +34,70 @@ def create(kobject_klass, attributes:, customer:)
       response.success? || raise(SolidusKustomer::CreateError)
     end
 
-    # Finds the customer via the Solidus database id
+    # Finds a customer via their Spree::User record. If the user does not have a kustomer_id
+    # associated, it also updates the original record to cache the UUID.
     #
-    # @param user [Spree::User] the user whose data we're looking for
+    # @param user [Spree::User] the customer we're looking for
     #
-    # @return [Hash] the customer data
-    def find_customer_by_external_id(user)
+    # @return [Hash] the customer data or nil if not present
+    def find_customer(user)
+      return find_customer_by_uuid(user.kustomer_id) if user.kustomer_id
+
+      customer = find_customer_by_email(user.email) || find_customer_by_external_id(user.id)
```

Comment:
Wouldn't it be better to attempt to find the customer by their external ID first? The external ID will never change, while a customer's email may change over time.

Reply:
I've had the same feeling, although I went for email first because in case of migrating from another system, the external ID may already have been set on Kustomer, with a very low chance of being it the same on Solidus. With that in mind I chose the other piece of information that would match even in this case.

In case of new users this wouldn't be an issue because the UUID would be present anyway. What do you think of my reasoning?

Reply:
That makes sense, I think. Thanks for explaining it!